### PR TITLE
Add eval_detection_coco

### PR DESCRIPTION
**Pull request by yuyu2172**
# Adding `pycocotools` as dependency

Since evaluation code is tricky to reimplement, I think it is better to use existing code base.
This situation is different from the existing evaluation scripts whose originals are written in MATLAB.

### CODE REVIEW

**Comment by yuyu2172** [comment]
@Hakuyume 
Any comments?

File: chainercv/evaluations/eval_detection_coco.py [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+import itertools
+import numpy as np
+import os
+import six
+import sys
+
+try:
+    import pycocotools.coco
+    import pycocotools.cocoeval
+    _availabel = True
```
Comment (Hakuyume) [inline chainercv/evaluations/eval_detection_coco.py]: typo `available`

**Comment by Hakuyume** [comment]
It is OK to use existing tool if users can install it easily.

**Comment by yuyu2172** [comment]
Please review

File: .travis.yml [hunk 2]
```diff
@@ -32,6 +32,7 @@ install:
       export LD_LIBRARY_PATH="$HOME/miniconda/envs/chainercv/lib:$LD_LIBRARY_PATH";
       conda env create -f environment.yml;
       source activate chainercv;
+      (cd ..; pip install -e 'git+https://github.com/pdollar/coco.git#egg=pycocotools&subdirectory=PythonAPI')
```
Comment (Hakuyume) [inline .travis.yml]: Do we need `cd`?

File: chainercv/evaluations/eval_detection_coco.py [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+        The :math:`l`-th value corresponds to the average precision \
+        for class :math:`l`. If class :math:`l` does not exist in \
+        either :obj:`pred_labels` or :obj:`gt_labels`, the corresponding \
+        value is set to :obj:`numpy.nan`.
+    .. [#coco_det_eval_2] An array of average recalls. \
+        The :math:`l`-th value corresponds to the average precision \
+        for class :math:`l`. If class :math:`l` does not exist in \
+        either :obj:`pred_labels` or :obj:`gt_labels`, the corresponding \
+        value is set to :obj:`numpy.nan`.
+    .. [#coco_det_eval_3] The average of Average Precisions over classes.
```
Comment (Hakuyume) [inline chainercv/evaluations/eval_detection_coco.py]: This line uses `Average Precisions`, where the other lines uses `average precisions/average recalls`

File: .travis.yml (hunk 2 above)
Comment (yuyu2172) [reply .travis.yml]: I didn't want to have anything under `chainercv/`.

Comment (Hakuyume) [reply .travis.yml]: Does it download something into current directory?

Comment (yuyu2172) [reply .travis.yml]: Yes.

**Pull request merged**